README.rst

```diff
@@ -2,56 +2,132 @@
+
+This demo seeks to obtain reasonable results for a nurse schedule, based on
+INH's model. Our implementation attempts to find a schedule for a number ``n_nurses`` of nurses and a number ``n_days`` of days that satisfies the following conditions:
+
+* One, and only one, nurse has been assigned to each day (hard shift 
+  constraint)
+* No nurse works two days in a row (hard nurse constraint)
+* The nurses should work the same number of days
+
+Running the demo results in the following output, at the command-line:
+::
+  Size  33
+
+  Energy  0.5999999999999694
+
+  Checking Hard nurse constraint  0.0
+
+  Checking Hard shift constraint  0.0
+
+  Checking Soft nurse constraint  0.6
+
+========= =  =  =  =  =  =  =  =  =  =  ==
+   Day    0  1  2  3  4  5  6  7  8  9  10
+========= =  =  =  =  =  =  =  =  =  =  ==
+Nurse  0           X     X     X     X   
+Nurse  1     X        X     X           X
+Nurse  2  X     X                 X      
+========= =  =  =  =  =  =  =  =  =  =  ==
+
+The results show the following:
+
+* One, and only one, nurse has been assigned to each day
+* No nurse works two days in a row
+* Two nurses work 4 days, and one works three days. Because two nurses work
+  one extra day each, the soft nurse constraint energy is nonzero. Each nurse 
+  working one extra day contributes a total of gamma to the energy. Since
+  gamma is 0.3, the total energy is expected to be 0.6.
 
 Usage
 -----
 
-A simple command that runs the demo. For example,
+To run the demo, run the command
 
 .. code-block:: bash
 
-  python <demo_name>.py
+  python nurse_scheduling.py
 
 
 Code Overview
 -------------
 
-A general overview of how the code works.
+Here is a general overview of the Nurse Scheduling code:
 
-Prefer bite-sized descriptions in bullet points:
-
-* Here's an example bullet point
+* Assign the size of the problem (number of nurses and days) and parameters
+* Compute the "penalty matrix" J
+* Develop the QUBO matrix
+* Run the problem (solve the QUBO)
+* Calculate the hard nurse constraint sum, to check if the hard nurse
+  constraints are satisfied
+* Calculate the hard shift constraint sum, to check if the hard shift
+  constraints are satisfied
+* Calculate the soft nurse constraint sum, to check if the soft nurse
+  constraints are satisfied
+* Print the nurse schedule
 
+Note that the total of the three constraint sums should equal the energy.
 
 Code Specifics
 --------------
 
-Notable parts of the code implementation.
+Some notes on the code:
 
-This is the place to:
+* We use a one-dimensional vector to store the two-dimensional (nurse, day) 
+  QUBO matrix. The (nurse, day) tuples are stored in the one-dimensional
+  vector in the following order, where nurse is first index, and day is 
+  second index, in the tuples:
 
-* Highlight a part of the code implementation
-* Talk about unusual or potentially difficult parts of the code
-* Explain a code decision
+  (0, 0) (0, 1) (0, 2)... (0, D) (1, 0) (1, 1)... (1, D)
```

Comment:
It's not clear to me where this "one-dimensional vector" exists in your code. I do not see any `(nurse, day) tuples .. stored in the one-dimensional vector`. Could you please point out this vector in your code?

I had initially thought the use of vectors and tuples as a typo and that you had really meant the indices of the QUBO matrix, but I'm now not so sure.

Reply:
So, the indices are used, for example, in the Q matrix, which goes as Q[ind, ind], where 'ind' is a composite index from (nurse, day). The QUBO has 2 of those indices.
How would you suggest changing the README?

Reply:
I see. I think there could be potential for confusion over the "two-dimensional (nurse, day) QUBO matrix" and the combined Q matrix (as they're both QUBO matrices). Perhaps you could add something saying that this one dimensional vector is used as an index? Similar to what you have in your above comment?

Reply:
@m3ller, I have added text. Please have a look. Hopefully this will do the trick.